mkosi/__init__.py

```diff
@@ -3229,10 +3227,8 @@ def run_finalize_script(config: MkosiConfig, state: MkosiState, for_cache: bool)
     verb = "build" if state.do_run_build_script else "final"
 
     with complete_step("Running finalize script…"):
-        env = collections.ChainMap(dict(BUILDROOT=str(state.root), OUTPUTDIR=str(output_dir(config))),
-                                   config.environment,
-                                   os.environ)
-        run([config.finalize_script, verb], env=env)
+        run([config.finalize_script, verb],
+            env=dict(BUILDROOT=str(state.root), OUTPUTDIR=str(output_dir(config)), **config.environment))
```

Comment:
Thinking some more about this, I think this version is slightly better.
```suggestion
            env={**config.environment), "BUILDROOT": str(state.root), "OUTPUTDIR": str(output_dir(config))} 
```

This couldn't happen before (because of the `ChainMap`), but now if there's already a `BUILDROOT` or `OUTPUTDIR` in the `config.environment` this would throw a `TypeError` because the same keyword argument is passed multiple times. 

Doing it in dict literal is slightly more efficient (because `dict` could be overwritten by a custom function, while the literal can't, so less lookups) and it doesn't have this problem, because the last value for a key wins, but then we need to unpack the `config.environment` first, so that our values win.